src/tree/FunctionAppTreeItem.ts

```diff
@@ -18,18 +18,18 @@ export class FunctionAppTreeItem implements IAzureParentTreeItem {
     public static contextValue: string = 'azFuncFunctionApp';
     public readonly contextValue: string = FunctionAppTreeItem.contextValue;
     public readonly siteWrapper: SiteWrapper;
+    public state: string | undefined;
 
     private readonly _functionsTreeItem: FunctionsTreeItem;
     private readonly _appSettingsTreeItem: AppSettingsTreeItem;
-    private readonly _state: string | undefined;
     private readonly _outputChannel: OutputChannel;
 
     public constructor(site: Site, outputChannel: OutputChannel) {
         this.siteWrapper = new SiteWrapper(site);
         if (!site.state) {
             throw new ArgumentError(site);
         }
-        this._state = site.state;
+        this.state = site.state;
```

Comment:
Don't quite get the usage here.  Is site.state obtained from Azure when the wrapper is originally created?  Perhaps it would be better to require using getState always?

Reply:
Yes, site.state is the state from when the instance of 'site' is originally created. I think it makes sense to use that in the constructor of the node? Using 'getState' feels redundant in this case to me...

Also we can't _always_ use getState because it doesn't give use the in-between values of 'starting' or 'stopping'